hazelcast/include/hazelcast/client/connection/Connection.h

```diff
@@ -70,7 +69,6 @@ class HAZELCAST_API Connection
                spi::ClientContext& client_context,
                int32_t connection_id,
                internal::socket::SocketFactory& socket_factory,
-               ClientConnectionManagerImpl& client_connection_manager,
```

Comment:
If a public API needs to be changed, there should be a real reason. As Connection is public method, we should not remove the parameter to remove the warning. We might something like this (void)client_connection_manager; to suppress the warning.